nio/client/async_client.py

```diff
@@ -2378,13 +2379,16 @@ def provider(got_429, got_timeouts):
     @client_session
     async def download(
         self,
-        server_name: str,
-        media_id: str,
+        server_name: Optional[str] = None,
+        media_id: Optional[str] = None,
+        mxc: Optional[str] = None,
         filename: Optional[str] = None,
         allow_remote: bool = True,
     ) -> Union[DownloadResponse, DownloadError]:
         """Get the content of a file from the content repository.
 
+        Either `server_name` and `media_id` or `mxc` must be given.
```

Comment:
I'd reword this, based on what poljar said, to say that `server_name` and `media_id` are going to be deprecated and that `mxc` is the way to go.